src/main/java/com/squareup/javapoet/TypeSpec.java

```diff
@@ -234,6 +242,10 @@ void emit(CodeWriter codeWriter, String enumName, Set<Modifier> implicitModifier
     }
   }
 
+  public boolean isAnnotationDeclaration() {
```

Comment:
The responsibility of `hasModifier` isn't the same. `hasModifier` checks `modifiers` field of `TypeSpec`. `isAnnotationDeclaration` checks `kind` field of `TypeSpec`. But we can have a new method like :

```java
public boolean isKind(Kind kind) {
  return this.kind == kind;
}
```

What do you think of it?